crates/utils/re_mcap/src/layers/ros2.rs

```diff
@@ -28,15 +28,19 @@ impl MessageLayer for McapRos2Layer {
         channel: &mcap::Channel<'_>,
         num_rows: usize,
     ) -> Option<Box<dyn MessageParser>> {
-        let Some(name) = channel.schema.as_ref().map(|schema| schema.name.as_str()) else {
+        let Some(schema) = channel.schema.as_ref() else {
             re_log::warn_once!(
                 "Encountered ROS2 message without schema in channel {:?}",
                 channel.topic
             );
             return None;
         };
 
-        Some(match name {
+        if schema.encoding.as_str() != "ros2msg" {
+            return None;
```

Comment:
should there be a warning for unsupported encoding or did that already happen elsewhere?

Reply:
Yeah, the warning happens one level above this because the individual layers can't know which other layers are active / supported.